test/frontend/util/get-jira-configuration-helpers.test.ts

```diff
@@ -9,88 +16,214 @@ describe("getFailedConnections", () => {
+
+			await RepoSyncState.create({
+				subscriptionId: 12345678,
+				repoId: 1,
+				repoName: "github-for-jira",
+				repoOwner: "atlassian",
+				repoFullName: "atlassian/github-for-jira",
+				repoUrl: "github.com/atlassian/github-for-jira",
+				repoUpdatedAt: new Date(0)
+			});
+			const result = await getFailedConnections(singleFailedInstallation, singleSubscription);
+			expect(result).toHaveLength(1);
+			expect(result).toEqual([{ deleted: true, id: 12345678, orgName: "atlassian" }]);
+		});
 	});
 });
+
+describe("getInstallation", () => {
+	let subscription: SubscriptionClass;
+	const gitHubInstallationId = 1234;
+	let match;
+
+	beforeEach(async () => {
+		match = {
+			syncStatus: "IN PROGRESS",
+			syncWarning: null,
+			// subscriptionUpdatedAt: new Date(0), // Not testing this as it's using moment
+			totalNumberOfRepos: 2,
+			numberOfSyncedRepos: 1,
+			jiraHost
+		};
+
+		subscription = await Subscription.create({
+			gitHubInstallationId,
+			jiraHost,
+			syncStatus: "ACTIVE",
+			repoSyncState: {
+				installationId: gitHubInstallationId,
+				jiraHost,
+				numberOfSyncedRepos: 1,
+				repos: {
+					"1": {
+						pullStatus: "complete",
+						branchStatus: "complete",
+						commitStatus: "complete",
+						lastBranchCursor: "foo",
+						lastCommitCursor: "bar",
+						lastPullCursor: 12,
+						repository: {
+							id: "1",
+							name: "github-for-jira",
+							full_name: "atlassian/github-for-jira",
+							html_url: "github.com/atlassian/github-for-jira",
+							owner: {
+								login: "atlassian"
+							},
+							updated_at: new Date(0)
+						}
+					},
+					"2": {
+						branchStatus: "failed",
+						repository: {
+							id: "2",
+							name: "github-for-jira",
+							full_name: "atlassian/github-for-jira",
+							html_url: "github.com/atlassian/github-for-jira",
+							owner: {
+								login: "atlassian"
+							},
+							updated_at: new Date(0)
+						}
+					}
+				}
+			}
+		});
+
+		githubNock
+			.get(`/app/installations/${gitHubInstallationId}`)
+			.reply(200, {
+				"id": gitHubInstallationId,
+				"account": {
+					"login": "atlassian",
+					"id": 1,
+					"node_id": "MDQ6VXNlcjE=",
+					"avatar_url": "https://github.com/images/error/octocat_happy.gif",
+					"url": "https://api.github.com/users/octocat",
+					"html_url": "https://github.com/octocat",
+					"type": "User",
+					"site_admin": false
+				},
+				"html_url": "https://github.com/organizations/github/settings/installations/1",
+				"app_id": 1,
+				"target_id": 1,
+				"target_type": "Organization"
+			});
+	});
+
+	afterEach(async () => {
+		await Subscription.destroy({ truncate: true });
+		await RepoSyncState.destroy({ truncate: true });
+	});
+
+	it("Should get installation from repoSyncState", async () => {
+		const result = await getInstallation(GitHubAPI(), subscription);
+		expect(result).toMatchObject(match);
+	});
+
+	it("Should get installation from Repo Sync State table", async () => {
```

Comment:
Nice to see that the tests for the old and new logic both assert against the same expected object.